vividus-plugin-rest-api/src/test/java/org/vividus/bdd/steps/api/HttpRequestStepsTests.java

```diff
@@ -149,6 +155,18 @@ void testSetCustomRequestConfigNoField()
         assertEquals("nonExistentField", exception.getMessage());
     }
 
+    @Test
+    void testWaitForResponseCode()
+    {
+        int retryTimes = 5;
+        int responseCode = 200;
+        SubSteps stepsToExecute = mock(SubSteps.class);
+        when(httpTestContext.getResponse())
+                .thenReturn(new HttpResponse());
+        httpRequestSteps.waitForResponseCode(responseCode, Duration.ofSeconds(10), retryTimes, stepsToExecute);
+        verify(stepsToExecute, times(retryTimes)).execute(Optional.empty());
+    }
```

Comment:
missing tests for cases:
- response == null
- response.code == expected response code